kernel/kernel-api/src/test/scala/io/delta/kernel/internal/files/ParsedLogDataSuite.scala

```diff
@@ -0,0 +1,378 @@
+    val parsed = ParsedLogData.forFileStatus(fileStatus)
+
+    assert(parsed.isInstanceOf[ParsedCheckpointData])
+    assert(parsed.version == 10)
+    assert(parsed.`type` == ParsedLogType.CLASSIC_CHECKPOINT)
+    assert(parsed.getCategory == ParsedLogCategory.CHECKPOINT)
+    assert(parsed.isMaterialized)
+    assert(!parsed.isInline)
+    assert(parsed.getFileStatus == fileStatus)
+  }
+
+  test("Classic checkpoint file equality") {
+    val cp1 = ParsedLogData.forFileStatus(classicCheckpointFileStatus(10))
+    val cp2 = ParsedLogData.forFileStatus(classicCheckpointFileStatus(10))
+    val cp3 = ParsedLogData.forFileStatus(classicCheckpointFileStatus(11))
+
+    assert(cp1 == cp1)
+    assert(cp1 == cp2)
+    assert(cp1 != cp3)
+  }
+
+  test("Correctly parses V2 checkpoint file") {
+    val fileStatus = v2CheckpointFileStatuses(Seq((20, true, 0)), "parquet").head._1
+    val parsed = ParsedLogData.forFileStatus(fileStatus)
+
+    assert(parsed.isInstanceOf[ParsedCheckpointData])
+    assert(parsed.version == 20)
+    assert(parsed.`type` == ParsedLogType.V2_CHECKPOINT)
+    assert(parsed.getCategory == ParsedLogCategory.CHECKPOINT)
+    assert(parsed.isMaterialized)
+    assert(!parsed.isInline)
+    assert(parsed.getFileStatus == fileStatus)
+  }
+
+  test("V2 checkpoint file equality") {
+    val fileStatus1 = v2CheckpointFileStatuses(Seq((20, false, 0)), "parquet").head._1
+    val fileStatus2 = v2CheckpointFileStatuses(Seq((20, false, 0)), "parquet").head._1
+    val fileStatus3 = v2CheckpointFileStatuses(Seq((21, true, 0)), "parquet").head._1
+
+    val parsed1 = ParsedLogData.forFileStatus(fileStatus1)
+    val parsed2 = ParsedLogData.forFileStatus(fileStatus2)
+    val parsed3 = ParsedLogData.forFileStatus(fileStatus3)
+
+    assert(parsed1 == parsed1)
+    assert(parsed1 == parsed2)
+    assert(parsed1 != parsed3)
+  }
+
+  ////////////////////////////
+  // Multi-Part Checkpoints //
+  ////////////////////////////
+
+  test("Inline multi-part checkpoint") {
+    ParsedMultiPartCheckpointData.forInlineData(10, 1, 3, emptyColumnarBatch)
+  }
+
+  test("Throws on inline multi-part checkpoint using wrong factory") {
+    val exMsg1 = intercept[IllegalArgumentException] {
+      ParsedLogData.forInlineData(10, ParsedLogType.MULTIPART_CHECKPOINT, emptyColumnarBatch)
+    }.getMessage
+    assert(exMsg1 ==
+      "For MULTIPART_CHECKPOINT, use ParsedMultiPartCheckpointData.forInlineData() instead")
+
+    val exMsg2 = intercept[IllegalArgumentException] {
+      ParsedCheckpointData.forInlineData(10, ParsedLogType.MULTIPART_CHECKPOINT, emptyColumnarBatch)
+    }.getMessage
+    assert(exMsg2 ==
+      "For MULTIPART_CHECKPOINT, use ParsedMultiPartCheckpointData.forInlineData() instead")
+  }
+
+  test("Correctly parses multi-part checkpoint file") {
+    val chkpt_15_1_3 = multiPartCheckpointFileStatus(15, 1, 3)
+    val parsed = ParsedLogData
+      .forFileStatus(chkpt_15_1_3)
+      .asInstanceOf[ParsedMultiPartCheckpointData]
```

Comment:
Do we expect to do this sort of casting throughout our code? Doesn't need to be addressed in this PR just wondering if that's how we plan to use these other more specific methods or if there is another solution

Reply:
Casting will only ever need to be done _once_, and it will be done in snapshot manager, when we parse the filestatuses into ParsedLogData and then group by the category